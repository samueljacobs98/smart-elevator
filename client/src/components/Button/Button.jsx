import "./Button.scss";

const Button = ({ floorNumber, onClick, isDisabled }) => {
  return (
    <button
      onClick={onClick}
      className={`button ${isDisabled ? "button--disabled" : ""}`}
      value={floorNumber}
      aria-label={`Call lift to floor ${floorNumber}`}
    >
      {floorNumber}
    </button>
  );
};

export default Button;
