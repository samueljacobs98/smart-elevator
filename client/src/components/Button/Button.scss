@use "../../assets/scss/variables" as *;

.button {
  width: 4rem;
  height: 4rem;
  border-radius: 50%;
  background: $colour-gradient-radial-blue;
  border: 1px solid black;
  text-align: center;
  line-height: 4rem;
  cursor: pointer;
  transition: box-shadow 0.2s ease-in-out;

  &:hover {
    box-shadow: 0px 0px 2px 2px rgba(0, 0, 0, 0.25);
  }

  &:active {
    box-shadow: 0px 0px 2px 2px rgba(155, 155, 155, 0.25);
  }

  @media (min-width: $breakpoint-mobile) {
    width: 5rem;
    height: 5rem;
  }

  @media (min-width: $breakpoint-tablet) {
    width: 6rem;
    height: 6rem;
  }
}
