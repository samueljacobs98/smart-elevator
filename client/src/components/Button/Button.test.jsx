import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import Button from "./Button";

describe("Button", () => {
  const mockOnClick = jest.fn();

  const renderComponent = (floorNumber = 0, isDisabled = false) =>
    render(
      <Button
        floorNumber={floorNumber}
        onClick={mockOnClick}
        isDisabled={isDisabled}
      />
    );

  it("renders without crashing", () => {
    renderComponent();

    expect(screen).toBeDefined();
  });

  it("displays the correct floor number", () => {
    const floorNumber = 5;

    renderComponent(floorNumber);

    const button = screen.getByRole("button");

    expect(button).toHaveTextContent(floorNumber);
  });

  it("applies the disabled class when isDisabled is true", () => {
    renderComponent(0, true);

    const button = screen.getByRole("button");

    expect(button).toHaveClass("button--disabled");
  });

  it("does not apply disabled class when isDisabled is false", () => {
    renderComponent(0, false);

    const button = screen.getByRole("button");

    expect(button).not.toHaveClass("button--disabled");
  });

  it("has the correct aria-label", () => {
    const floorNumber = 5;

    renderComponent(floorNumber);

    const button = screen.getByRole("button");

    expect(button).toHaveAttribute(
      "aria-label",
      `Call lift to floor ${floorNumber}`
    );
  });

  it("calls onClick function when clicked", () => {
    renderComponent();

    const button = screen.getByRole("button");

    fireEvent.click(button);

    expect(mockOnClick).toHaveBeenCalled();
  });

  it("should match snapshot", () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });
});
