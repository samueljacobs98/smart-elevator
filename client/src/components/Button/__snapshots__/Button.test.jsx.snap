// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<div>
  <button
    aria-label="Call lift to floor 0"
    class="button "
    value="0"
  >
    0
  </button>
</div>
`;
